src/Microsoft.AspNetCore.Mvc.ViewFeatures/Rendering/TagBuilder.cs

```diff
@@ -196,16 +196,17 @@ private void AppendAttributes(TextWriter writer, HtmlEncoder encoder)
                 foreach (var attribute in Attributes)
                 {
                     var key = attribute.Key;
+                    var value = attribute.Value ?? string.Empty;
```

Comment:
We could move this to Line 209. Avoids the null - coalescing if we aren't going to ignore it for `id` attribute.